src/main/java/com/ericsson/ei/services/SubscriptionService.java

```diff
@@ -93,26 +107,43 @@ public boolean doSubscriptionExist(String name) {
     }
     
     @Override
-    public boolean modifySubscription(Subscription subscription, String subscriptionName) {
+    public boolean modifySubscription(Subscription subscription, String subscriptionName, String userName) {
         ObjectMapper mapper = new ObjectMapper();
         try {
             String StringSubscription = mapper.writeValueAsString(subscription);
             String subscriptionNameQuery = String.format(SUBSCRIPTION_NAME, subscriptionName);
+ 
+            if(!userName.isEmpty()){  
+                String queryUser = String.format(USER_NAME, userName);      
+                String operatorAnd = "{$and:[%s]}";
+                String queryTemp = subscriptionNameQuery+ "," + queryUser;            
+                subscriptionNameQuery = String.format(operatorAnd, queryTemp);            
+            }
+            
+            
             return subscriptionRepository.modifySubscription(subscriptionNameQuery, StringSubscription);
         } catch (JsonProcessingException e) {
             return false;
         }
     }
     
     @Override
-    public boolean deleteSubscription(String name) {
+    public boolean deleteSubscription(String name, String userName) {
         String query = String.format(SUBSCRIPTION_NAME, name);
+        if(!userName.isEmpty()){  
+            String queryUser = String.format(USER_NAME, userName);      
+            String operatorAnd = "{$and:[%s]}";
+            String queryTemp = query+ "," + queryUser;            
+            query = String.format(operatorAnd, queryTemp);            
+        }
```

Comment:
same comment here as well.
And it looks like to be exactly same code in all place. Break it out to own function and reuse the code where its needed in the code.